src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpHandlerResourceV3Provider.cs

```diff
@@ -105,17 +116,44 @@ public CredentialPromptWebRequestHandler(IWebProxy proxy, ICredentials credentia
                         if (ProxyAuthenticationRequired(ex) &&
                             HttpHandlerResourceV3.PromptForProxyCredentials != null)
                         {
-                            // prompt use for proxy credentials.
-                            var credentials = await HttpHandlerResourceV3
-                                .PromptForProxyCredentials(request.RequestUri, Proxy, cancellationToken);
+                            ICredentials currentCredentials = Proxy.Credentials;
 
-                            if (credentials == null)
+                            try
                             {
-                                throw;
+                                await _credentialPromptLock.WaitAsync();
+
+                                // Check if the credentials have already changed
+                                if (beforeAuthId != _lastAuthId)
+                                {
+                                    continue;
+                                }
+
+                                // Limit the number of retries
+                                _authRetries++;
+                                if (_authRetries >= MaxAuthRetries)
```

Comment:
I don't actually see any difference in how we establish the numbers, and we end up with random number spread through the code base. Lets consolidate into one and call it a day